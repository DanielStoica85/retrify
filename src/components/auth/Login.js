import React, { Component } from 'react';
import LabeledFormInput from '../layout/LabeledFormInput';
import { Button, Container, Form, FormGroup, Alert } from 'reactstrap';
import { connect } from 'react-redux';
import { login } from '../../actions/auth';
import { Redirect } from 'react-router-dom';

class Login extends Component {
    state = {
        email: '',
        password: ''
    };

    handleChange = e => {
        this.setState({
            [e.target.id]: e.target.value
        });
    };

    handleSubmit = e => {
        e.preventDefault();
        this.props.login(this.state);
    };

    render() {
        const { authError, auth } = this.props;
        if (auth.uid) return <Redirect to="/" />;
        return (
            <Container>
                <Form onSubmit={this.handleSubmit}>
                    <h4 className="text-black">Sign In</h4>
                    <LabeledFormInput
                        labelText="Email"
                        labelFor="email"
                        inputType="email"
                        inputName="email"
                        inputId="email"
                        inputPlaceholder="Add your email..."
                        handleChange={this.handleChange}
                    />
                    <LabeledFormInput
                        labelText="Password"
                        labelFor="password"
                        inputType="password"
                        inputName="password"
                        inputId="password"
                        inputPlaceholder="Add your password..."
                        handleChange={this.handleChange}
                    />
                    <FormGroup>
                        <Button color="primary">Login</Button>
                    </FormGroup>
                    {authError ? (
                        <Alert color="danger">{authError}</Alert>
                    ) : null}
                </Form>
            </Container>
        );
    }
}

const mapStateToProps = state => {
    return {
        authError: state.auth.authError,
        auth: state.firebase.auth
    };
};

const mapDispatchToProps = dispatch => {
    return {
        login: credentials => dispatch(login(credentials))
    };
};

export default connect(
    mapStateToProps,
    mapDispatchToProps
)(Login);
