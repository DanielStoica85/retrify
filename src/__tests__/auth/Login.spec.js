import React from 'react';
import renderer from 'react-test-renderer';
import Login from '../../components/auth/Login';

describe('Login component', () => {
    it('renders correctly', () => {
        const component = renderer.create(<Login />).toJSON();
        expect(component).toMatchSnapshot();
    });
});
