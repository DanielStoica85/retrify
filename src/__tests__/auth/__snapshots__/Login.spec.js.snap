// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login component renders correctly 1`] = `
<div
  className="container"
>
  <form
    className=""
    onSubmit={[Function]}
  >
    <h4
      className="text-black"
    >
      Sign In
    </h4>
    <div
      className="form-group"
    >
      <label
        className=""
        htmlFor="email"
      >
        Email
      </label>
      <input
        className="form-control"
        id="email"
        name="email"
        onChange={[Function]}
        placeholder="Add your email..."
        type="email"
      />
    </div>
    <div
      className="form-group"
    >
      <label
        className=""
        htmlFor="password"
      >
        Password
      </label>
      <input
        className="form-control"
        id="password"
        name="password"
        onChange={[Function]}
        placeholder="Add your password..."
        type="password"
      />
    </div>
    <div
      className="form-group"
    >
      <button
        aria-label={null}
        className="btn btn-primary"
        onClick={[Function]}
      >
        Login
      </button>
    </div>
  </form>
</div>
`;
